sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,196 @@
+                                          self._hand_serial, [dated_file])
+        rospy.sleep(1)
+
+        if os.path.exists(self._log_file_path + dated_file):
+            os.remove(self._log_file_path + dated_file)
+
+        return success
+
+    def _download_from_AWS(self):
+        success = False
+        success = self.aws_manager.download("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                            self._hand_serial, [self._get_latest_file_name_from_AWS()])
+        if success:
+            rospy.sleep(1)
+            latest_file_name_from_AWS = self._get_latest_file_name_from_AWS()
+            shutil.copy(self._log_file_path + latest_file_name_from_AWS, self._log_file_path + self._log_file_name)
+            rospy.sleep(1)
+            if os.path.exists(self._log_file_path + latest_file_name_from_AWS):
+                os.remove(self._log_file_path + latest_file_name_from_AWS)
+
+        return success
+
+    def _save_data_localy(self, event):
+        if self._verify_data_empty():
+            print("saving data locally")
+            self._complete_data['total_angles_[rad]'] = self._current_values
+            self._complete_data['total_time_[s]'] = rospy.get_rostime().secs
+            f = open(self._log_file_path+self._log_file_name, 'w')
+            yaml.safe_dump(self._complete_data, f)
+            f.close()
+
+    def _verify_data_empty(self):
+        if len(self._current_values.keys()) == 0:
+            return False
+        return True
+
+    def _load_data_from_yaml(self):
+        with open(self._log_file_path+self._log_file_name, 'r') as f:
+            self._complete_data = yaml.load(f, Loader=yaml.SafeLoader)
+        self._current_values = self._complete_data['total_angles_[rad]']
+        self._current_time = self._complete_data['total_time_[s]']
+
+    def _update_with_higher_values(self, local_data, aws_data):
+        for k, v in local_data.items():
+            if isinstance(v, dict):
+                local_data[k] = self._update_with_higher_values(aws_data.get(k, {}), v)
+            else:
+                local_data[k] = max(local_data[k], aws_data[k])
+        return local_data
+
+    def _update_log(self, local_file_path, aws_file_path):
+        with open(local_file_path, 'r') as f_local:
+            data_local = yaml.load(f_local, Loader=yaml.SafeLoader)
+        with open(aws_file_path, 'r') as f_aws:
+            data_aws = yaml.load(f_aws, Loader=yaml.SafeLoader)
+
+        self._complete_data = self._update_with_higher_values(data_local, data_aws)
+        self._save_data_localy(None)
+
+    def _init_log(self):
+
+        if not os.path.exists(self._log_file_path):
+            os.makedirs(self._log_file_path)
+            self._download_from_AWS()
+
+        if os.path.exists(self._log_file_path + self._log_file_name):
+            rospy.loginfo("Comparing!")
```

Comment:
remove this or add more useful log info message

Reply:
Will be removed in the final version